_threshold_maps_ratio changes the input map

My maps images keep changing when I use RegionExtractor. I think we need to make a copy [here](https://github.com/nilearn/nilearn/blob/master/nilearn/regions/region_extractor.py#L58)
For instance the following code throws an `AssertionError: Arrays are not equal`
```Python
from nilearn._utils.data_gen import generate_maps
import numpy as np
from nilearn.regions.region_extractor import _threshold_maps_ratio

maps, mask_img = generate_maps((10, 10, 10), 30)
maps.get_data()[:5] = 100
maps_data = maps.get_data().copy()
thresholded_maps = _threshold_maps_ratio(maps, threshold=1)
np.testing.assert_array_equal(maps.get_data(), maps_data)

```

--- a/nilearn/regions/region_extractor.py
+++ b/nilearn/regions/region_extractor.py
@@ -55,7 +55,7 @@ def _threshold_maps_ratio(maps_img, threshold):
     else:
         ratio = threshold
 
-    maps_data = _safe_get_data(maps, ensure_finite=True)
+    maps_data = _safe_get_data(maps, ensure_finite=True).copy()
 
     abs_maps = np.abs(maps_data)
     # thresholding

--- a/nilearn/regions/tests/test_region_extractor.py
+++ b/nilearn/regions/tests/test_region_extractor.py
@@ -51,10 +51,15 @@ def test_threshold_maps_ratio():
     # smoke test for function _threshold_maps_ratio with randomly
     # generated maps
 
+    # Check that the input maps are not changed
+    maps, _ = generate_maps((6, 8, 10), n_regions=3)
+    maps.get_data()[:3] = 100
+    maps_data = maps.get_data().copy()
+    thr_maps = _threshold_maps_ratio(maps, threshold=1.0)
+    np.testing.assert_array_equal(maps.get_data(), maps_data)
+
     # make sure that n_regions (4th dimension) are kept same even
     # in thresholded image
-    maps, _ = generate_maps((6, 8, 10), n_regions=3)
-    thr_maps = _threshold_maps_ratio(maps, threshold=1.0)
     assert_true(thr_maps.shape[-1] == maps.shape[-1])
 
     # check that the size should be same for 3D image

--- a/nilearn/regions/tests/test_region_extractor.py
+++ b/nilearn/regions/tests/test_region_extractor.py
@@ -51,8 +51,9 @@ def test_threshold_maps_ratio():
     # smoke test for function _threshold_maps_ratio with randomly
     # generated maps
 
-    # Check that the input maps are not changed
     maps, _ = generate_maps((6, 8, 10), n_regions=3)
+
+    # test that there is no side effect
     maps.get_data()[:3] = 100
     maps_data = maps.get_data().copy()
     thr_maps = _threshold_maps_ratio(maps, threshold=1.0)

